csharp/private/wrappers/dotnet.cc

```diff
@@ -50,14 +63,34 @@ int main(int argc, char** argv) {
       evprintf("DOTNET_CLI_TELEMETRY_OPTOUT", "1"),  // disable telemetry
   };
 
+  auto extra_count =
+    !!(std::strlen(kArgv1) != 0) +
+    !!(std::strlen(kArgv2) != 0);
+
+  auto dotnet_argv = new char*[extra_count + argc + 1];
+
   // dotnet wants this to either be dotnet or dotnet.exe but doesn't have a
   // preference otherwise.
-  auto dotnet_argv = new char*[argc + 1];
-  dotnet_argv[0] = (char*)"dotnet";
+  dotnet_argv[0] = const_cast<char*>("dotnet");
+
+  // Make sure to hold a reference to these std::string.
+  auto argv1 = runfiles->Rlocation(kArgv1);
```

Comment:
https://github.com/bazelbuild/bazel/blob/46b285a8044d8c2dcf292f40b6b74bd99d66b430/tools/python/runfiles/runfiles.py#L138

is `""` `not`y?

Reply:
https://github.com/bazelbuild/bazel/blob/46b285a8044d8c2dcf292f40b6b74bd99d66b430/tools/cpp/runfiles/runfiles_src.cc#L170